### PR TITLE
Add OpenAPI IDL BBEs

### PR DESCRIPTION
**Pull request by lnash94**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/openapi-idl-client/openapi_idl_client.bal [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import ballerina/openapi;
+import ballerina/io;
+// Add the client import with OpenAPI specification.
+client "openapi.yaml" as foo;
+
+configurable string weatherAPIKey = ?;
+
+// You can use `@openapi:ClientConfig` annotation for customized your client.
```
Comment (azinneera) [inline examples/openapi-idl-client/openapi_idl_client.bal:8]:
```suggestion
// You can use the `@openapi:ClientConfig` annotation for customizing your client.
```

File: examples/openapi-idl-client/openapi_idl_client.bal [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+@openapi:ClientConfig {
+    tags: ["Current Weather Data"]
+}
+client "https://raw.githubusercontent.com/ballerina-platform/openapi-connectors/main/openapi/openweathermap/openapi.yaml" as weather;
```
Comment (azinneera) [inline examples/openapi-idl-client/openapi_idl_client.bal:12]: Shall we add the link of a specific tag/commit? The file in master may change.

Comment (lnash94) [reply examples/openapi-idl-client/openapi_idl_client.bal:12]: +1 , will update link with specific tag/commit 😃

**Comment by lnash94** [comment]
Close this PR in favour of this PR https://github.com/ballerina-platform/ballerina-dev-website/pull/5750

**Pull request closed without merging**